docs/reference/developer/tooling.rst

```diff
@@ -14,7 +14,7 @@ Note: Ubuntu users can use the `Kitware Repo <https://apt.kitware.com/>`_ to obt
 Compilers
 ~~~~~~~~~
 
-- GCC >= 7
+- GCC >= 9
 - LLVM Clang >= 7
```

Comment:
I haven't tried with LLVM clang in some time. This is probably also stale now (version 7) and would be better to remove altogether.
I'm using xcode 12.4, as are the build servers. I'm pretty sure we can no longer use 10 or 11. so that should be updated too.
We also don't list MSVC here, but that is supported.